Stop child component's onClick event when Slider is dragged.

lonyele opened the pull request:
Hi, I recently have been using this library, it worked fine for my use cases except this behaviour of triggering child component's onClick event after Slide is dragged or swiped. Similar problems that have been issued here are #848, #1160, #542, #604    

User of the library can manually block onClick event by making a dragging state on their own way using beforeChange(), afterChange(), I think this can be implemented in library by default, because I found this is a common feature that many people would want it right away using this library

My solution is so simple that It adds one state called `previouslyDragged` boolean and 
on this.swipeStart(), sets `previouslyDragged` to false
on this.swipeMove(), sets `previouslyDragged` to true
on onClickCapture() event, stop propagation if previous state was dragged or swiped

Now, just clicking each item of slide works as expected and after the slider is dragged or swiped It won't trigger child component's onClick event

rafbc commented:
Any ETA to when this will be merged?! :P

young666 commented:
This doesn't work when I use <a> instead of <div> in Slider, I have try disable drag and drop for <a>, still doesn't work 😟

rafbc commented:
I used this
```
constructor(props) {
		super(props);

		this.state = {
			previouslyDragged: false
		}
	}

	handleImageClick(e) {
		const { previouslyDragged } = this.state;
		if (previouslyDragged) {
			e.stopPropagation();
			return;
		} else {
			console.log('CLICK FINALLY WORKS')
		}

	}
```


```

		const settings = {
			dots: true,
			infinite: true,
			speed: 500,
			arrows: true,
			slidesToShow: 1,
			slidesToScroll: 1,
			responsive: [
				{
					breakpoint: 768,
					settings: {
						slidesToShow: 1
					}
				},
				{
				breakpoint: 480,
					settings: {
						slidesToShow: 1
					}
				}
			],
			beforeChange: () => {
				this.setState({ previouslyDragged: true })
			},
			afterChange: () => {
				this.setState({ previouslyDragged: false })
			}
		};

<Carousel {...settings}>
....
</Carousel>
```

young666 commented:
@rafbc it only works if after drag I move my mouse away from the item I dragged 😃

lonyele commented:
@young666 You are right. I can fix <a> tag issue by adding "event.preventDefault()" to where I added "event.stopPropagation()".  What I need was to block events thus I put only stopPropagation, but If there is a need to stop browser's default actions then preventDefault() is also necessary.

After I added preventDefault(), I found <a> tag's link doesn't work when dragged. However It is really janky and still I can drag text of <a> tag's link.  After I set the draggable={false} on <a> tag, text is not draggable.

I added an example at the bottom so that you can check the behaviour of each cases

akashmdkml commented:
@lonyele  Hey nice work, I tried your fix, now it does not fire the click event when I drag or swipe. but now the drag feature is hard to drag experience, especially when I swipe from left to right it does not swipe properly. let me know if you have any suggestion for this. Thanks. 
here is my demo code https://codesandbox.io/s/myqq0yz54y

deleterepo commented:
Any updates on this one? I see the changes were approved but not merged yet.

lonyele commented:
Sadly, it seems like this project is not maintained anymore(nothings get merged recently)

raphaelbs commented:
Any chance this will get merged?

lonyele commented:
@akiran just a gentle ping...
